src/hooks/chart.ts

```diff
@@ -1432,42 +1461,93 @@ export const useChartData = (plot: Plot) => {
+    //CHECK: Flag to set if we want to show specific invalid configuration alerts. If it's false general alert will be shown.
+    let showAlert = false;
+    //If the plot has multiple traces, multiTrace will be set to true
+    const multiTrace = unpackedResponses.length > 1;
+    result.data = unpackedResponses.map((responseData: ResponseData, index: number) => {
       const currTrace = plot.traces[index];
+      //To add trace number against the alert message if multiple traces are given for a plot
+      const alertMsg = multiTrace ? `Trace ${index + 1}: ` : '';
       hovertemplate_display_pattern = currTrace.hovertemplate_display_pattern; //use trace info
-      if (plot.plot_type === 'bar') {
-        // returns an array of objects similar to PlotlyPlotData
-        const barPlotData = parseBarResponse(currTrace, plot, responseData);
-        barPlotData.forEach((data: any) => plotlyData.push(data));
-      } else if (plot.plot_type === 'pie') {
-        // returns a single object similar to PlotlyPieData
-        plotlyData.push(parsePieResponse(currTrace, plot, responseData));
-      } else if (plot.plot_type === 'scatter') {
-        // returns a single object similar to PlotlyPlotData
-        plotlyData.push(parseScatterResponse(currTrace, plot, responseData));
-      } else if (plot.plot_type === 'histogram') {
-        // returns an array of objects similar to PlotlyPlotData
-        const histogramPlotData = parseHistogramResponse(currTrace, plot, responseData);
-        histogramPlotData.forEach((data: any) => plotlyData.push(data))
-      } else if (plot.plot_type === 'violin') {
-        const { result: parseResult, layout } = parseViolinResponse(
-          currTrace,
-          plot,
-          responseData,
-          selectDataGrid,
-          { width, height },
-          templateParams.noData
-        );
-        additionalLayout = layout;
-        plotlyData.push(parseResult);
-      } else if (plot.plot_type === 'heatmap') {
-        const heatmapData = parseHeatmapResponse(currTrace, plot, responseData);
-        additionalLayout = { ...heatmapData.layoutParams };
-        plotlyData.push(heatmapData.result);
+      //If the response_format is configured then check the format against type of file and parse the data accordingly
+      if (responseData && currTrace.response_format) {
+        //If the given format is not from the allowed types then show an alert warning
+        if (!(validFileTypes.includes(currTrace.response_format)) && !alertFunctions.alerts.some(
+          (alert) => alert.message.includes(invalidResponseFormatAlert))) {
+          alertFunctions.addAlert(alertMsg + invalidResponseFormatAlert, ChaiseAlertType.WARNING);
+        }
+        //If the given format is csv and the content of file is also of type csv then parse the data using csv parser
+        else if (currTrace.response_format === 'csv' && !isDataJSON(responseData)) {
```

Comment:
`isDataJson(responseData)` is being called 4 times in this function for different conditionals. This should be done once outside of the `if (responseData && currTrace.response_format)` conditional since the check won't change. Then it can be reused in all 4 places.